### PR TITLE
Add recipe for consensify

### PR DESCRIPTION
**Pull request by dramanica**
This PR adds a recipe for consensify, a method for generating a consensus pseudohaploid genome sequence

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

## Summary by CodeRabbit

- **New Features**
	- Introduced a new shell script for building the `consensify` project.
	- Added package configuration for `consensify`, version `2.4.0`, including metadata and dependencies.

- **Documentation**
	- Included detailed package information, such as source URL, checksum, and installation verification commands.

### CODE REVIEW

File: recipes/consensify/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+## change to source dir
+cd ${SRC_DIR}
```
[the remark this answers is not in the document]
Comment (dramanica) [reply recipes/consensify/build.sh]: fixed as requested

File: recipes/consensify/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('consensify', max_pin="x") }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - zlib
```
[the remark this answers is not in the document]
Comment (dramanica) [reply recipes/consensify/meta.yaml:20]: `zlib` is necessary, as it is linked to from the c++ code. Removing it causes the building step to fail.

File: recipes/consensify/build.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+## change to source dir
+cd "${SRC_DIR}" || exit 1
```
Comment (martin-g) [inline recipes/consensify/build.sh:4]: No need. It is already in this folder

File: recipes/consensify/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+test:
+  source_files:
+    - test/**
+  commands:
+    - consensify_c -h
+    - consensify_c -c ./test/eg_missingness.counts.gz -p ./test/eg_missingness.pos.gz -s ./test/scaffold_lengths.txt -o ./test/test_gz.fasta -seed 123
+
+about:
+  home: https://github.com/jlapaijmans/Consensify
+  license: GPL3
```
Comment (martin-g) [inline recipes/consensify/meta.yaml:33]:
```suggestion
  license: GPL-3.0-or-later
```

File: recipes/consensify/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  license_file: LICENCE.txt
```
Comment (martin-g) [inline recipes/consensify/meta.yaml:34]:
```suggestion
  license_file: LICENCE.txt
  license_family: GPL3
```

File: recipes/consensify/build.sh (hunk 3 above)
Comment (dramanica) [reply recipes/consensify/build.sh]: Thank you! Removed as per suggestion.

File: recipes/consensify/meta.yaml (hunk 4 above)
Comment (dramanica) [reply recipes/consensify/meta.yaml]: Thanks, updated licence as suggested.

File: recipes/consensify/meta.yaml (hunk 5 above)
Comment (dramanica) [reply recipes/consensify/meta.yaml:32]: Thanks, updated as suggested.

File: recipes/consensify/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+about:
+  home: https://github.com/jlapaijmans/Consensify
+  license: GPL-3.0-or-later
+  license_family: GPL
+  license_file: LICENCE.txt
+  dev_url: https://github.com/jlapaijmans/Consensify
+  doc_url: https://github.com/jlapaijmans/Consensify
+  summary: A method for generating a consensus pseudohaploid genome sequence
+  about: Consensify is a method for generating a consensus pseudohaploid genome sequence with greatly reduced error rates compared to standard pseudohaploidisation. The method is described in full and tested in the associated publication (Barlow et al. (2020), Genes 11:50, doi 10.3390/genes11010050).
```
[the remark this answers is not in the document]
Comment (dramanica) [reply recipes/consensify/meta.yaml:36]: This suggestion uses a different family grouping than the one suggested by martin-g and will be ignored.

**Comment by dramanica** [comment]
@BiocondaBot please add label

**Review by apeltzer — APPROVED** [review]

**Pull request merged**